app/src/main/java/edu/mills/heartoftaiwanese/activity/favorites/FavoritesContract.kt

```diff
@@ -0,0 +1,19 @@
+package edu.mills.heartoftaiwanese.activity.favorites
+
+import edu.mills.heartoftaiwanese.data.DatabaseWord
+
+interface FavoritesContract {
+    interface FavoritesView {
+        fun onConfigurationSuccess()
+        fun onWordListChanged(newWordList: List<DatabaseWord>)
+    }
+
+    interface FavoritesViewModel {
+        fun favoriteWord(word: DatabaseWord)
+
+        /**
+         * Get the updated word list. Return true if there are more than 0 words, false if there are none.
```

Comment:
The name makes one expect that this will return a list, not a boolean. My guess is that you combined updating with checking the size. It's better to have clear code.